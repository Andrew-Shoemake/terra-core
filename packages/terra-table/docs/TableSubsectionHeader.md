# Terra Table Subsection Header

The Terra Subsection Header is built upon a table row and can be placed similarly within a table, provided the number of columns are provided to the SubsectionHeader.

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-table`
  - `yarn add terra-table`

## Usage

```jsx
import React from 'react';
import { SubsectionHeader } from 'terra-table';

<SubsectionHeader
  key="unique-react-key"
  isCollapsible
  isCollapsed
  metaData={{ myInfo: 'my-data' }}
  onSelect={this.handleSelection}
  colSpan={3}
/>
```

## Component Features
* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
