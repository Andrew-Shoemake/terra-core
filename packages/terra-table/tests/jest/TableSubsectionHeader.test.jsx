import React from 'react';
import TableSubsectionHeader from '../../src/TableSubsectionHeader';

// Snapshot test
it('should render a default table section header', () => {
  const section = shallow(
    <TableSubsectionHeader colSpan={1} title="test" />,
  );
  expect(section).toMatchSnapshot();
});


it('should render a table section header with isCollpased', () => {
  const section = shallow(
    <TableSubsectionHeader colSpan={1} title="test" isCollapsed />,
  );
  expect(section).toMatchSnapshot();
});

it('should render a table section header with isCollapsible', () => {
  const section = shallow(
    <TableSubsectionHeader colSpan={1} title="test" isCollapsible />,
  );
  expect(section).toMatchSnapshot();
});

it('should render a table section header with level', () => {
  const section = shallow(
    <TableSubsectionHeader colSpan={1} title="test" level={4} />,
  );
  expect(section).toMatchSnapshot();
});

it('should render a table section header with metaData', () => {
  const section = shallow(
    <TableSubsectionHeader colSpan={1} title="test" metaData={{}} />,
  );
  expect(section).toMatchSnapshot();
});

it('should render a table section header with onSelect', () => {
  const section = shallow(
    <TableSubsectionHeader colSpan={1} title="test" onSelect={jest.fn()} />,
  );
  expect(section).toMatchSnapshot();
});

it('should render a table section header with refCallback', () => {
  const section = shallow(
    <TableSubsectionHeader colSpan={1} title="test" onSelect={jest.fn()} />,
  );
  expect(section).toMatchSnapshot();
});
