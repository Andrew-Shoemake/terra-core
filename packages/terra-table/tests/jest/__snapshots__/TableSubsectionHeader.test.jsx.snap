// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a default table section header 1`] = `
<tr
  className="subsection-header"
>
  <td
    colSpan={1}
  >
    <div
      className="subsection-content"
    >
      <h2
        className="title"
      />
    </div>
  </td>
</tr>
`;

exports[`should render a table section header with isCollapsible 1`] = `
<tr
  aria-expanded={true}
  aria-level={2}
  className="subsection-header is-collapsible"
  data-row-show-focus="true"
  onBlur={[Function]}
  onMouseDown={[Function]}
  role="heading"
  tabIndex="0"
>
  <td
    colSpan={1}
  >
    <div
      className="subsection-content"
    >
      <div
        className="start"
      >
        <span
          className="accordion-icon is-open"
        />
      </div>
      <div
        className="fill"
      >
        <h2
          className="title"
        />
      </div>
    </div>
  </td>
</tr>
`;

exports[`should render a table section header with isCollpased 1`] = `
<tr
  className="subsection-header"
>
  <td
    colSpan={1}
  >
    <div
      className="subsection-content"
    >
      <h2
        className="title"
      />
    </div>
  </td>
</tr>
`;

exports[`should render a table section header with level 1`] = `
<tr
  className="subsection-header"
>
  <td
    colSpan={1}
  >
    <div
      className="subsection-content"
    >
      <h2
        className="title"
      />
    </div>
  </td>
</tr>
`;

exports[`should render a table section header with metaData 1`] = `
<tr
  className="subsection-header"
>
  <td
    colSpan={1}
  >
    <div
      className="subsection-content"
    >
      <h2
        className="title"
      />
    </div>
  </td>
</tr>
`;

exports[`should render a table section header with onSelect 1`] = `
<tr
  className="subsection-header"
>
  <td
    colSpan={1}
  >
    <div
      className="subsection-content"
    >
      <h2
        className="title"
      />
    </div>
  </td>
</tr>
`;

exports[`should render a table section header with refCallback 1`] = `
<tr
  className="subsection-header"
>
  <td
    colSpan={1}
  >
    <div
      className="subsection-content"
    >
      <h2
        className="title"
      />
    </div>
  </td>
</tr>
`;

exports[`should render a table section header with title 1`] = `
<tr
  className="subsection-header"
>
  <td
    colSpan={1}
  >
    <div
      className="subsection-content"
    >
      <h2
        className="title"
      >
        test
      </h2>
    </div>
  </td>
</tr>
`;
